File: C#/Clock/Clock.cs
using System;

public class Clock
{
    public static int Modulus(int a, int b) {
        return ((a % b) + b) % b;
    }

    private int _hours;
    private int _minutes;

    public Clock(int hours, int minutes)
    {
        var quotient = (int)Math.Floor((double)minutes / 60);
        var remainder = Modulus(minutes, 60);

        this._hours = Modulus((hours + quotient), 24);
        this._minutes = remainder;
    }

    public Clock Add(int minutesToAdd)
    {
        return new Clock(this._hours, this._minutes + minutesToAdd);
    }

    public Clock Subtract(int minutesToSubtract)
    {
        return new Clock(this._hours, this._minutes - minutesToSubtract);
    }

    public override string ToString()
    {
        return $"{this._hours.ToString().PadLeft(2, '0')}:{this._minutes.ToString().PadLeft(2, '0')}";
    }

    public override bool Equals(object obj)
    {
        if (obj is Clock other)
        {
            return this.GetHashCode() == other.GetHashCode();
        }

        return false;
    }

    public override int GetHashCode()
    {
        return this._hours.GetHashCode() ^ this._minutes.GetHashCode();
    }
}

File: JavaScript/clock.js
function modulus(a, b) {
    return ((a % b) + b) % b;
}

class Clock {
    constructor(hour, minute) {
        this.hour = modulus(hour, 24);

        if (minute !== undefined) {
            var quotient = Math.floor(minute / 60);
            var remainder = modulus(minute, 60);

            this.hour = modulus((hour + quotient), 24);
            this.minute = remainder;
        }
        else {
            this.minute = 0;
        }
    }

    toString() {
        return `${this.hour.toString().padStart(2, "0")}:${this.minute.toString().padStart(2, "0")}`;
    }

    plus(minutes) {
        return new Clock(this.hour, this.minute + minutes);
    }

    minus(minutes) {
        return new Clock(this.hour, this.minute - minutes);
    }

    equals(other) {
        return this.hour === other.hour && this.minute === other.minute;
    }
}

let c = new Clock(-1, 15);
console.log(c.toString());